cdb2jdbc/src/main/java/com/bloomberg/comdb2/jdbc/DatabaseDiscovery.java

```diff
@@ -44,13 +45,12 @@ public class DatabaseDiscovery {
     static final String CDB2DBCONFIG_SECONDARY_PATH = "/bb/etc/cdb2/config.d/";
 
     /* comdb2lcldb, per jvm. */
-    static class TimeAndHosts {
+    private static final class TimeAndHosts {
         long ms;
-        ArrayList<String> hosts = new ArrayList<String>();
+        List<String> hosts = new ArrayList<>();
     }
-    static HashMap<String, TimeAndHosts> comdb2lcldb = new HashMap<String, TimeAndHosts>();
-    /* Make it visible to us only. */
-    private static Object lock = new Object();
+
+    private static final Map<String, TimeAndHosts> comdb2lcldb = new ConcurrentHashMap<>();
```

Comment:
I believe it's safer to use the map type that's specifically been designed with concurrency in mind. At the moment there is an unsynchronized call to `comdb2lcldb.remove` in `static void getDbHosts`.